client/rest-high-level/src/test/java/org/elasticsearch/client/SearchIT.java

```diff
@@ -513,7 +513,7 @@ public void testSearchWithSuggest() throws IOException {
         assertNull(searchResponse.getAggregations());
         assertEquals(Collections.emptyMap(), searchResponse.getProfileResults());
         assertEquals(0, searchResponse.getHits().totalHits);
-        assertEquals(0f, searchResponse.getHits().getMaxScore(), 0f);
+        assertEquals(Float.NaN, searchResponse.getHits().getMaxScore(), 0f);
         assertEquals(0, searchResponse.getHits().getHits().length);
         assertEquals(1, searchResponse.getSuggest().size());
```

Comment:
I am not too happy that all the failures I got came from integration tests and docs tests. I added some assertions to `QueryPhaseTests` but maybe that is not enough. Suggestions are welcome on where to add tests for this change.